conans/client/conf/required_version.py

```diff
@@ -0,0 +1,28 @@
+from conans.client.cache.cache import ClientCache
+from conans.client.graph.range_resolver import satisfying
+from conans import __version__ as client_version
+from conans.errors import ConanException
+
+
+def check_required_conan_version(cache_folder, out):
+    """ Check if the required Conan version in config file matches to the current Conan version
+
+            When required_conan_version is not configured, it's skipped
+            When required_conan_version is configured, Conan's version must matches the required
+            version
+            When it doesn't match, an ConanException is raised
+
+        :param cache_folder: Conan cache folder
+        :param out: Output stream
+        :return: None
+    """
+    cache = ClientCache(cache_folder, out)
+    required_version = cache.config.required_conan_version
+    if required_version:
+        output = ""
+        result = satisfying([client_version], required_version, output)
```

Comment:
IMO we shouldn't couple this functionality to the way we resolve version ranges. It's unlikely that we are going to change it soon, but there was an issue with the proposal. We can still use the same python library, but here we only need one line:

```python
from semver import satisfies

true/false = satisfies(client_version, required_version)    
   
``` 

Do you think so?

Reply:
if it works, I agree that is better! Thanks @jgsogo